Tasks/BashV3/bash.ts

```diff
@@ -72,7 +74,19 @@ async function run() {
                 targetFilePath = input_filePath;
             }
 
-            contents = `. '${targetFilePath.replace("'", "'\\''")}' ${input_arguments}`.trim();
+            if (old_source_behavior) {
+                contents = `. '${targetFilePath.replace("'", "'\\''")}' ${input_arguments}`.trim();
+            } else {
+                // Check if executable bit is set
+                let stats: fs.Stats = fs.statSync(targetFilePath);
+                if ((stats.mode & 1) > 0) {
+                    contents = `sh '${targetFilePath.replace("'", "'\\''")}' ${input_arguments}`.trim();
```

Comment:
is the purpose to run the script in a sub-shell? Should we use 'bash' instead of 'sh' since 'sh' is not guaranteed to the 'bash' on all systems.

Reply:
Yeah, that's a good catch, updated appropriately